src/core/obj.js

```diff
@@ -32,578 +32,600 @@ function fetchDestination(dest) {
+      }).then(function(parent) {
+        if (!parent) {
+          return null;
+        }
+        if (!isDict(parent)) {
+          throw new FormatError('Parent must be a dictionary.');
+        }
+        return parent.getAsync('Kids');
+      }).then(function(kids) {
+        if (!kids) {
+          return null;
+        }
+
+        const kidPromises = [];
+        let found = false;
+        for (let i = 0, ii = kids.length; i < ii; i++) {
+          const kid = kids[i];
+          if (!isRef(kid)) {
+            throw new FormatError('Kid must be a reference.');
           }
-          return parent.getAsync('Kids');
-        }).then(function (kids) {
-          if (!kids) {
-            return null;
+          if (isRefsEqual(kid, kidRef)) {
+            found = true;
+            break;
           }
-          var kidPromises = [];
-          var found = false;
-          for (var i = 0; i < kids.length; i++) {
-            var kid = kids[i];
-            if (!isRef(kid)) {
-              throw new FormatError('kid must be a Ref.');
+          kidPromises.push(xref.fetchAsync(kid).then(function(kid) {
+            if (!isDict(kid)) {
+              throw new FormatError('Kid node must be a dictionary.');
             }
-            if (isRefsEqual(kid, kidRef)) {
-              found = true;
-              break;
+            if (kid.has('Count')) {
+              total += kid.get('Count');
+            } else { // Page leaf node.
+              total++;
             }
-            kidPromises.push(xref.fetchAsync(kid).then(function (kid) {
-              if (!isDict(kid)) {
-                throw new FormatError('kid node must be a Dict.');
-              }
-              if (kid.has('Count')) {
-                var count = kid.get('Count');
-                total += count;
-              } else { // page leaf node
-                total++;
-              }
-            }));
-          }
-          if (!found) {
-            throw new FormatError('kid ref not found in parents kids');
-          }
-          return Promise.all(kidPromises).then(function () {
-            return [total, parentRef];
-          });
+          }));
+        }
+        if (!found) {
+          throw new FormatError('Kid reference not found in parent\'s kids.');
+        }
+        return Promise.all(kidPromises).then(function() {
+          return [total, parentRef];
         });
-      }
+      });
+    }
 
-      var total = 0;
-      function next(ref) {
-        return pagesBeforeRef(ref).then(function (args) {
-          if (!args) {
-            return total;
-          }
-          var count = args[0];
-          var parentRef = args[1];
-          total += count;
-          return next(parentRef);
-        });
-      }
+    let total = 0;
+    function next(ref) {
+      return pagesBeforeRef(ref).then(function(args) {
+        if (!args) {
+          return total;
+        }
+        const count = args[0];
+        const parentRef = args[1];
```

Comment:
You should be able to use array destructuring instead, i.e. `const [count, parentRef] = args;`.